### DISCUSSION TITLE
Precompiled Firmware - BLTouch offset

### DISCUSSION THREAD

**Discussion by Xploder** [Jyers/Marlin#76] [Q&A] [answered]
What's the offset for the BLTouch sensor in the precompiled Firmwares?

In the BLTouch Template configuration.h you define it with
#define NOZZLE_TO_PROBE_OFFSET { -10, -10, 0 }

Is that the offset for the official metal bracket Creality provides?

**Comment by t3chguy** [comment, marked as answer]
You can specify the offset in the UI under Advanced settings

**Reply by Xploder** [reply]
You can adjust the X and Y Offset in UI?
Sorry it's been a while since I've used Marlin. I've been using RepRap for a while now and back when I've used Marlin these values had to be set in the config file.

**Reply by t3chguy** [reply]
Yes, in Jyers DWIN rewrite it should be under Prepare -> Advanced I believe.

**Reply by Xploder** [reply]
That's awesome. Thanks for your answer.

**Comment by Deletedice** [comment]
@Xploder What was the x offset that you ended up with?

**Reply by Xploder** [reply]
I'll be able to measure it tomorrow but I've read that it's -44 for X and -6 for Y for the official creality bracket

**Reply by Schnabulation** [reply]
I have the official Creality metal bracket and have mine measured at -45 and -7.

Z-offset is at -3 in my setup.

**Reply by Xploder** [reply]
We could load the official Creality BLTouch firmware and get the offset by sending M851, right?

**Reply by Deletedice** [reply]
> I have the official Creality metal bracket and have mine measured at -45 and -7.
> 
> Z-offset is at -3 in my setup.

This is what i got, just wanted to compare, thanks!

**Reply by danrough** [reply]
I've just run into the problem of the probe missing the left hand side of the bed while probing to create the mesh. So firstly, thanks for taking the time to report this and to post your answers. That's already been helpful :) Secondly though, I'd like to understand how you arrived at the numbers above - would one of you mind giving me some pointers? I've read a few guides, all of which include something about setting the z offset, but none of which have noted that setting the x y offset might also be a requirement. I'm wondering what I missed, and how I might arrive at those numbers for myself - I like to understand these things :) Thanks!

**Reply by danrough** [reply]
When I create a mesh, from about point 60 on the grid upwards, even using the numbers you have above, the probe is missing the bed by a considerable margin.

**Reply by VashBaldeus** [reply]
> 
> 
> I have the official Creality metal bracket and have mine measured at -45 and -7.
> 
> Z-offset is at -3 in my setup.

I think that the X/Y are same for anyone who uses the Creality bracket that comes with genuine BL Touch, the Z depends on bed shape really. -3 did not work well for me, -3.2 works better for me
